src/nci_util.c

```diff
@@ -308,6 +308,140 @@ nci_parse_iso_dep_poll_a_param(
     return FALSE;
 }
 
+static
+gboolean
+nci_parse_nfc_dep_poll_param(
+    NciActivationParamNfcDepPoll* param,
+    const guint8* bytes,
+    guint len)
+{
+    /* ATR_RES Length */
+    const guint8 atr_res_len = bytes[0];
+
+    /*
+     * NFCForum-TS-NCI-1.0
+     * Table 82: Activation Parameters for NFC-DEP Poll Mode
+     *
+     * +=========================================================+
+     * | Offset | Size | Description                             |
+     * +=========================================================+
+     * | 0      | 1    | Length of ATR_RES Command Parameter (n) |
+     * | 1      | n    | ATR_RES bytes from and including Byte 3 |
+     * +=========================================================+
+     */
+    if (atr_res_len >= 15 && len >= atr_res_len + 1) {
+        const guint8* atr_res = bytes + 1;
+
+        /*
+         * NFCForum-TS-DigitalProtocol-1.0
+         * 14.6.3 ATR_RES Response
+         */
+        memcpy(param->nfcid3, atr_res, sizeof(param->nfcid3));
+        param->did = atr_res[10];
+        param->bs = atr_res[11];
+        param->br = atr_res[12];
+        param->to = atr_res[13];
+        param->pp = atr_res[14];
+        if (atr_res_len > 15) {
+            param->g.bytes = atr_res + 15;
+            param->g.size = atr_res_len - 15;
```

Comment:
Funtion to "parse from bytes" which then caches pointer to 'bytes' data -- is it ok / going to stay valid?

Reply:
elsewhere too e.g. in nci_parse_nfc_dep_listen_param()

Reply:
Yes, the idea is that those "parse" functions don't allocate memory - and they are allowed to use pointers to the memory block which was passed in. The caller is supposed to make sure that those pointers remain valid for as long as necessary.